xds/src/main/java/io/grpc/xds/WrrLocalityLoadBalancer.java

```diff
@@ -70,8 +84,14 @@ public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
               wrrLocalityConfig.childPolicy));
     }
 
-    childLb = wrrLocalityConfig.childPolicy.getProvider().newLoadBalancer(helper);
-    childLb.handleResolvedAddresses(
+    // If this is the first update or the new LB config has a different policy, switch over to it.
+    PolicySelection newPolicySelection = wrrLocalityConfig.childPolicy;
+    if (!newPolicySelection.getProvider().getPolicyName().equals(currentPolicyName)) {
```

Comment:
This optimization is handled by graceful switch. So just do the code in the block unconditionally. https://github.com/grpc/grpc-java/blob/f2348b0157cae21074a48d2c0428c9750fc51b2f/core/src/main/java/io/grpc/util/GracefulSwitchLoadBalancer.java#L117-L122